metabase/metabase
# fix z-issues on move to tab menu

### Description

This fixes two issues with zIndex related to the "move to tab" menu
First issue is a zindex issue with grid maps: https://metaboat.slack.com/archives/C057T1QTB3L/p1698169540830189
![move-to-tab-z-index](https://github.com/metabase/metabase/assets/1914270/c00eb8ae-bd01-4aad-92e4-26d2eeecbaf5)

Second issue is a zindex issue when editing a link card: https://github.com/metabase/metabase/pull/34961/files#r1371364752

<img width="458" alt="image" src="https://github.com/metabase/metabase/assets/1914270/7e80036c-e7fa-405c-a5b6-a22fffcf8480">



## Cause
Both issues were cause by me avoiding the portal for the menu for the first implementation.

I didn't use the portal because to avoid this:

https://github.com/metabase/metabase/assets/1914270/d0443929-4ea1-45bc-ac97-e3b17eba918a

Not using the portal made the menu a child of the dashcard actions panel, which is a child of the card, and so hovering the menu kept the the dashcard actions panel open. (Hovering the card, or having focus, keeps the actions panel visible)

That solution caused the two bugs above, I tried to solve them with via zIndex and no JS, but every thing that I fixed caused another zIndex issue so I stepped back and used the portal.

## Solution
In this PR I add a piece of state to check if there is an open sub menu, and in that case I keep the actions panel visible. 

I don't like that the state is currently in Dashcard, we could refactor `renderDashCardActions` to be a component so that we can keep the state internal to it.


### How to verify

1. Create/go to a dashboard with more than 1 tab
2. Create a link card, focus the input, the popover should open, hover the move to tab icon in the actions panel

For the second issue:
1. Create/go to a dashboard with more than 1 tab
2. Add a card with a map-> grid visualization
3. Hover the move to tab icon in the actions panel

### Demo


https://github.com/metabase/metabase/assets/1914270/fe965068-1b51-4b1e-bf14-bf0b2a84d483



### Checklist
